src/benchq/resource_estimation/_footprint_analysis.py

```diff
@@ -66,20 +68,42 @@ def estimate_cost(self) -> CostEstimate:
         distillation_area = (
             self.factory_count * self.magic_state_factory.physical_qubit_footprint
         )
-        rounds = int(
-            self.toffoli_count / self.factory_count * self.magic_state_factory.rounds
-        )
-        distillation_failure = (
-            self.magic_state_factory.failure_rate * self.toffoli_count
-        )
-        data_failure = (
-            self.proportion_of_bounding_box
-            * _topological_error_per_unit_cell(
-                self.logical_data_qubit_distance, gate_err=self.physical_error_rate
+        if self.T_count != 0 and self.toffoli_count == 0:
+            rounds = int(
+                self.T_count / self.factory_count * self.magic_state_factory.rounds
+            )
+            distillation_failure = self.magic_state_factory.failure_rate * self.T_count
+            data_failure = (
+                self.proportion_of_bounding_box
+                * _topological_error_per_unit_cell(
+                    self.logical_data_qubit_distance, gate_err=self.physical_error_rate
+                )
+                * logical_storage
+                * rounds
             )
-            * logical_storage
-            * rounds
-        )
+
+        elif self.toffoli_count != 0 and self.T_count == 0:
```

Comment:
There seems to be quite a bit of code duplication here (which I think might have been the culprit for the missing factor of 1/2 above that @akataba caught). Also, if I recall, we were discussing some ways to account for both types of gates. One way to reduce duplication and also take a step towards accommodating both gates is to do something like the following:
```
if self.T_count * self.toffoli_count != 0:
    raise ValueError("Cannot accommodate circuits having both T gates and Toffoli gates.")

rounds = int(
(self.toffoli_count+self.T_count/2) / self.factory_count * self.magic_state_factory.rounds
)
distillation_failure = self.magic_state_factory.failure_rate * gate_count
data_failure = (
    self.proportion_of_bounding_box
    * _topological_error_per_unit_cell(
        self.logical_data_qubit_distance, gate_err=self.physical_error_rate
    )
    * logical_storage
    * rounds
)
```
Then (I think!) in the case that we want to accommodate both, we can simply remove the if statement.

Reply:
gate_count = self.t_count/2 + self.toffoli count, right ?